lib/iris/fileformats/grib/message.py

```diff
@@ -160,7 +164,7 @@ def data(self):
                 shape = (grid_section['numberOfDataPoints'],)
             else:
                 shape = (grid_section['Nj'], grid_section['Ni'])
-            proxy = _DataProxy(shape, np.dtype('f8'), np.nan,
+            proxy = _DataProxy(shape, self.data_dtype, np.nan,
```

Comment:
Do you mean the `data_dtype` on `GribWrapper`? As I understand it `GribWrapper` is used for the now deprecated loader, and this is for the new one. I don't think they're ever used together.